mindsdb_native/libs/phases/model_analyzer/model_analyzer.py

```diff
@@ -152,3 +157,70 @@ def predict_wrapper(*args, **kwargs):
             self.transaction.hmd['probabilistic_validators'][col] = pickle_obj(pval)
 
         self.transaction.lmd['validation_set_accuracy'] = sum(overall_accuracy_arr)/len(overall_accuracy_arr)
+
+        # conformal prediction confidence estimation
+        self.transaction.lmd['stats_v2']['train_std_dev'] = {}
+        self.transaction.hmd['label_encoders'] = {}
+        self.transaction.hmd['icp'] = {}
+
+        for target in output_columns:
+            data_type = self.transaction.lmd['stats_v2'][target]['typing']['data_type']
+            data_subtype = self.transaction.lmd['stats_v2'][target]['typing']['data_subtype']
+            is_classification = data_type == DATA_TYPES.CATEGORICAL
+
+            fit_params = {'target': target,
+                          'all_columns': self.transaction.lmd['columns'],
+                          'columns_to_ignore': self.transaction.lmd['columns_to_ignore'] +
+                                               [col for col in output_columns if col != target]}
+
+            if is_classification:
+                if data_subtype != DATA_SUBTYPES.TAGS   :
+                    all_targets = [elt[1][target].values for elt in inspect.getmembers(self.transaction.input_data)
+                                   if elt[0] in {'test_df', 'train_df', 'validation_df'}]
+                    all_classes = np.unique(np.concatenate([np.unique(arr) for arr in all_targets]))
+
+                    enc = OneHotEncoder(sparse=False)
+                    enc.fit(all_classes.reshape(-1, 1))
+                    fit_params['one_hot_enc'] = enc
+                    self.transaction.hmd['label_encoders'][target] = enc
+                else:
+                    fit_params['one_hot_enc'] = None
+                    self.transaction.hmd['label_encoders'][target] = None
+
+                adapter = ConformalClassifierAdapter
+                nc_function = MarginErrFunc()  # better than IPS as we'd need the complete distribution over all classes
+                nc_class = ClassifierNc
+                icp_class = IcpClassifier
+
+            else:
```

Comment:
It seems like this should check for == Numerical, otherwise it will just end up failing for outputs that are timeseries, text, images... etc